src/ExtraValuesProvider.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Wikibase\DataModel;
+
+/**
+ * Provides extra values for the serialization of an object.
+ *
+ * @since 2.1
+ *
+ * @licence GNU GPL v2+
+ * @author Bene* < benestar.wikimedia@gmail.com >
+ */
+interface ExtraValuesProvider {
```

Comment:
Could also be ExtraValuesSerializer, tbh I don't care that much how this interface is called. Initially I called it FooBar because I didn't find a better way ^^